src/relion/dbmodel/modeltables.py

```diff
@@ -1,26 +1,14 @@
 import functools
+import itertools
 import re
 
-# import ispyb
 from ispyb import sqlalchemy
 
 # if we replace uuid with count do not include 0 in the count because it will break some bool checks for None
 
 
-class ProcessID:
-    def __init__(self, start):
-        self.id = start
-
-    def __call__(self):
-        old_id = self.id
-        self.id += 1
-        return old_id
-
-    def reset(self, start):
-        self.id = start
-
-
-PID = ProcessID(1)
+ProcessID = itertools.count(1)
+# should this be attached to the table?
```

Comment:
Does this need to be a global singleton or could it be a table property?

Reply:
As the Zocalo buffer table only has a single column for a UUID from the wrapper I think the value this produces should be unique across all tables for a given wrapper instance. It could be moved to be a property of the Relion project itself once more integration has been done

Reply:
:+1:
`ProcessID` is then probably not the best name

Reply:
Agreed. `WrapperID`?

Reply:
Done